components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/lib/ace-editor/ext/beautify/siddhi-rules.js

```diff
@@ -0,0 +1,418 @@
+            newLine = '\n',
+            tab = '\t',
+            isCodeEndingWithANewLine = false,
+            isValueContainsANewLine = false,
+            isInsideAPartition = false,
+            lastNewLineCharacterIsAtRow = 0,
+            currentTokenRowNumber = 0,
+            commonAnnotationLevel = 0,
+            noOfTabsInBeginOfNewLineInAnnotation = 0;
+
+        while (token !== null) {
+
+            // user added new line handling starts here
+            var tokenRow = iterator.getCurrentTokenRow();
+            if (tokenRow > currentTokenRowNumber) {
+                var rowDifference = tokenRow - currentTokenRowNumber;
+                if (rowDifference > 1) {
+                    // adding only one empty line
+                    if (code !== "") { // don't add newlines in the beginning of the file
+                        if (!_.endsWith(code, newLine)) {
+                            code += newLine + newLine;
+                            isCodeEndingWithANewLine = true;
+                        } else {
+                            code += newLine;
+                            isCodeEndingWithANewLine = true;
+                        }
+                    }
+                } else {
+                    if (code !== "") { // don't add newlines in the beginning of the file
+                        code += newLine;
+                        isCodeEndingWithANewLine = true;
+                    }
+                }
+                currentTokenRowNumber = tokenRow;
+                lastNewLineCharacterIsAtRow = tokenRow - 1;
+
+                // adding tabs spaces for annotations which user entered a piece of the annotation in the second line
+                // ex: @sink(newLine type = 'log') will be converted to @sink(newLine tabSpace type = 'log')
+                if (commonAnnotationLevel > 0) {
+                    noOfTabsInBeginOfNewLineInAnnotation = 0;
+                    for (var i = 0; i < commonAnnotationLevel; i++) {
+                        code += tab;
+                        noOfTabsInBeginOfNewLineInAnnotation += 1;
+                    }
+                }
+            } // user added new line handling ends here
+
+            if (!token) {
+                token = iterator.stepForward();
+                continue;
+            }
+
+            nextToken = iterator.stepForward();
+            value = token.value;
+
+            // trim unnecessary white spaces to a single space other than new lines
+            if (token.type === 'whitespace') {
+                // @sink(newLine tabSpace space type = 'log') will be converted to @sink(newLine tabSpace type = 'log')
+                if (commonAnnotationLevel > 0 && _.endsWith(code, tab)) {
+                    token.value = token.value.trim();
+                    value = '';
+                    // all the other whitespaces in the line will be converted to space
+                } else {
+                    token.value = token.value.trim();
```

Comment:
Can get this line out of the condition and simplify the if block as follows.
```
value = commonAnnotationLevel > 0 && _.endsWith(code, tab) ? '' : space;
```

Reply:
Corrected in commit e30b1f4